pkg/events/dispatcher.go

```diff
@@ -240,3 +250,19 @@ func (e EventHandler) Run(tickDuration time.Duration) {
 		}
 	}()
 }
+
+func (e EventHandler) isNewLeader(currentLeaderStatus bool) (bool, error) {
+	isLeader, err := leaderelection.IsLeader()
+	if err != nil {
+		return currentLeaderStatus, err
+	}
+
+	if isLeader != currentLeaderStatus {
+		if err := e.repo.EventsReset(e.context); err != nil {
+			e.log.WithError(err).Error("failed to reset events on new leader")
+			return isLeader, err
```

Comment:
Hva gjør vi med prosesser som kjører? For her kan vi ende opp med at en prosess holder på, også starter vi en ny prosess. Kanskje ikke et stort problem :thinking:

Reply:
Fra elector https://github.com/nais/elector?tab=readme-ov-file#elector:
>As long as the leader pod exists, it will continue to be leader. This means that a leader in CrashLoopBackoff state will still be leader.

Sånn jeg tolker det så er dette grunnen til feilen vi har i dag. Det er kun når leder podden dør at vi får et bytte. Og dersom leder-podden dør mens en event prosesseres vil eventen aldri endre state til completed/failed men bare være stuck i processing.